android/src/main/java/org/devio/rn/splashscreen/SplashScreen.java

```diff
@@ -23,13 +30,89 @@
 public class SplashScreen {
     private static Dialog mSplashDialog;
     private static WeakReference<Activity> mActivity;
+    private static boolean isVideoActive = false;
+    private static boolean isImageActive = false;
+
+    public static void showVideo(final Activity activity) {
+        if (activity == null) return;
+        if (mSplashDialog != null) return;
+        if (isImageActive || isVideoActive) return;
+        isVideoActive = true;
+
+        mActivity = new WeakReference<Activity>(activity);
+        activity.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                if (!activity.isFinishing()) {
+                    mSplashDialog = new Dialog(activity, R.style.SplashScreen_Fullscreen);
+                    mSplashDialog.setContentView(R.layout.video_view);
+                    mSplashDialog.setCancelable(false);
+
+                    Context context = activity.getApplicationContext();
+
+                    String videoPath = "android.resource://" + context.getPackageName() + "/" + R.raw.splashscreen;
+
+                    MediaMetadataRetriever retriever = new MediaMetadataRetriever();
+                    retriever.setDataSource(context, Uri.parse(videoPath));
+                    int videoWidth = Integer.valueOf(retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_WIDTH));
+                    int videoHeight = Integer.valueOf(retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_HEIGHT));
+                    retriever.release();
+
+
+
+                    VideoView videoView = (VideoView)mSplashDialog.findViewById(R.id.video_view);
+                    int viewHeight = videoView.getHeight();
+                    int viewWidth = videoView.getWidth();
+
+                    float ratioWidth = viewWidth / videoWidth;
+                    float ratioHeight = viewHeight / videoHeight;
+                    float fullWidth;
+                    float fullHeight;
+                    if (ratioWidth < ratioHeight ) {
+                        fullWidth = videoWidth * ratioWidth;
+                        fullHeight = videoHeight * ratioWidth;
+                    } else {
+                        fullWidth = videoWidth * ratioHeight;
+                        fullHeight = videoHeight * ratioHeight;
+                    }
+
+                    videoView.getLayoutParams().width = Math.round(fullWidth);
+                    videoView.getLayoutParams().height = Math.round(fullHeight);
```

Comment:
Need to set video width/height to preserve aspect ratio. If you know a better way please lmk